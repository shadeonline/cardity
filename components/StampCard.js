import React, { useState } from 'react';
import { StyleSheet, View, Image, Text, TouchableOpacity, Modal } from 'react-native';
import StampTaskModal from './StampTaskModal';
const StampCard = ({ rows, stampsPerRow, progress }) => {
  const [selectedStamp, setSelectedStamp] = useState(null);

  const handleStampPress = (stampKey) => {
    setSelectedStamp(stampKey);
    console.log(stampKey)
  };

  const renderStampRows = () => {
    const stamps = [];

    for (let i = 0; i < rows; i++) {
      const row = [];
      for (let j = 0; j < stampsPerRow; j++) {
        const stampKey = i * stampsPerRow + j;
        const isFilled = stampKey < progress;
        const isLastStamp = j === stampsPerRow - 1;

        if (isLastStamp) {
          // Add treasure chest image to the last stamp in each row
          row.push(
            <TouchableOpacity
              key={stampKey}
              id='reward'
              style={styles.stamp}
              onPress={() => handleStampPress(stampKey)}
            >
              <Image
                source={require('../assets/treasure-chest.png')}
                style={styles.crownImage}
              />
              {isFilled && (
                <Image source={require('../assets/stamp.png')} style={styles.stampImage} />
              )}
            </TouchableOpacity>
          );
        } else {
          row.push(
            // Add numbers to each stamp and render img on top if stamp is filled
            <TouchableOpacity
              key={stampKey}
              style={styles.stamp}
              onPress={() => handleStampPress(stampKey)}
            >
              <Text>{stampKey + 1}</Text>
              {isFilled && (
                <Image source={require('../assets/stamp.png')} style={styles.stampImage} />
              )}
            </TouchableOpacity>
          );
        }
      }
      stamps.push(
        <View key={i} style={styles.stampRow}>
          {row}
        </View>
      );
    }
    return stamps;
  };

  const handleCloseModal = () => {
    setSelectedStamp(null);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.header}>Stamps Collected!</Text>
      {renderStampRows()}
      <StampTaskModal
        selectedStamp={selectedStamp}
        progress={progress}
        onCloseModal={handleCloseModal} 
        stampsPerRow = {stampsPerRow}/>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    textAlign: 'center',
    fontSize: 25,
    paddingVertical: 10,
  },
  container: {
    width: '100%',
    paddingHorizontal: 10,
    paddingVertical: 20,
    justifyContent: 'center',
  },
  stampRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 1,
  },
  stamp: {
    position: 'relative',
    width: 35,
    height: 35,
    marginLeft: 15,
    marginRight: 15,
    marginTop: 12,
    borderRadius: 25,
    backgroundColor: 'lightgrey',
    justifyContent: 'center',
    alignItems: 'center',
  },
  crownImage: {
    width: 30,
    height: 30,
    resizeMode: 'contain',
  },
  stampImage: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: 30,
    height: 30,
    resizeMode: 'contain',
  },
});

export default StampCard;
