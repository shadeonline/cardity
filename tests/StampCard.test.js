import React from 'react';
import { render } from 'react-native-testing-library';
import StampCard from '../components/StampCard';

describe('StampCard Component', () => {
  it('matches snapshot', () => {
    // Mock data
    const maxRewardStamps = 10;
    const progress = 5;
    const rewards = {
      1: true,
      2: true,
      // Add more rewards as needed for testing
    };
    const task = "Complete a task to earn a reward";

    // Render the StampCard component with the mock data
    const { toJSON } = render(
      <StampCard
        maxRewardStamps={maxRewardStamps}
        progress={progress}
        rewards={rewards}
        task={task}
      />
    );

    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
