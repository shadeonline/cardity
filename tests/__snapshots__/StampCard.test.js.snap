// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StampCard Component matches snapshot 1`] = `
<View
  style={
    {
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 25,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Stamps Collected!
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
      <Text>
        1
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
      <Text>
        2
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        3
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        4
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        5
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 30,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "top": 0,
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        6
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        7
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        8
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        9
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 25,
          "height": 35,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 1,
          "position": "relative",
          "width": 35,
        }
      }
    >
      <Text>
        10
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</View>
`;
